Code/Tools/SceneAPI/SceneData/ManifestBase/SceneNodeSelectionList.cpp

```diff
@@ -112,7 +156,8 @@ namespace AZ
                     return;
                 }
 
-                serializeContext->Class<SceneNodeSelectionList, DataTypes::ISceneNodeSelectionList>()->Version(2)
+                serializeContext->Class<SceneNodeSelectionList, DataTypes::ISceneNodeSelectionList>()
+                    ->Version(3, &SceneNodeSelectionListVersionConverter)
```

Comment:
An explicit reflection of `AZStd::vector<AZStd::string>` is need to ensure the type is available when using the `DataElementNode::GetData` function in the version converter.
Container templates such as `AZStd::vector` are implicitly reflected by the [SerializeContext::ClassBuilder::Field](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h#L2006) through the SerializeGenericTypeInfo template.

Therefore the current code relies on another class reflecting a field of `AZStd::vector<AZStd::string>` in order for the [SerializeDataElementNode::GetDataHierarchy](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/SerializeContext.cpp#L3005-L3012) to look up the type.

Therefore a call to `serializeContext->RegisterGenericType<AZStd::vector<AZStd::string>>()` is need here.

Reply:
You can see an example in the AssetBundleManifest.cpp [Reflect](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/Asset/AssetBundleManifest.cpp#L41-L43) function